internal/webhook/manager.go

```diff
@@ -0,0 +1,479 @@
+func DesiredValidatingWebhookConfiguration(namespace string) *admissionregistrationv1.ValidatingWebhookConfiguration {
+	vwc := &admissionregistrationv1.ValidatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: ValidatingWebhookConfigurationName,
+			Annotations: map[string]string{
+				InjectCabundleAnnotation: "true",
+			},
+		},
+		Webhooks: []admissionregistrationv1.ValidatingWebhook{
+			newValidatingWebhook(
+				DatascienceclusterValidatorName,
+				namespace,
+				dscwebhook.ValidateDatascienceClusterPath,
+				[]admissionregistrationv1.RuleWithOperations{
+					{
+						Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.Create},
+						Rule: admissionregistrationv1.Rule{
+							APIGroups:   []string{gvk.DataScienceCluster.Group},
+							APIVersions: []string{gvk.DataScienceCluster.Version},
+							Resources:   []string{"datascienceclusters"},
+						},
+					},
+				},
+				nil,
+			),
+			newValidatingWebhook(
+				DscinitializationValidatorName,
+				namespace,
+				dsciwebhook.ValidateDscinitializationPath,
+				[]admissionregistrationv1.RuleWithOperations{
+					{
+						Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.Create, admissionregistrationv1.Delete},
+						Rule: admissionregistrationv1.Rule{
+							APIGroups:   []string{gvk.DSCInitialization.Group},
+							APIVersions: []string{gvk.DSCInitialization.Version},
+							Resources:   []string{"dscinitializations"},
+						},
+					},
+				},
+				nil,
+			),
+			newValidatingWebhook(
+				AuthValidatorName,
+				namespace,
+				authwebhook.ValidateAuthPath,
+				[]admissionregistrationv1.RuleWithOperations{
+					{
+						Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.Create, admissionregistrationv1.Update},
+						Rule: admissionregistrationv1.Rule{
+							APIGroups:   []string{gvk.Auth.Group},
+							APIVersions: []string{gvk.Auth.Version},
+							Resources:   []string{"auths"},
+						},
+					},
+				},
+				nil,
+			),
+		},
+	}
+
+	vwc.Webhooks = append(vwc.Webhooks, getDesiredKueueValidatingWebhooks(namespace, KueueManagedLabelKey, "")...)
+	// TODO: Remove this once we drop support for the legacy label
+	vwc.Webhooks = append(vwc.Webhooks, getDesiredKueueValidatingWebhooks(namespace, KueueLegacyManagedLabelKey, "-legacy")...)
+
+	return vwc
+}
+
+// createOrUpdateWebhookConfiguration handles the creation or update of Mutating/ValidatingWebhookConfiguration.
+//
+// Parameters:
+//   - ctx: The context for the API call
+//   - c: The client to use for the API call
+//   - log: The logger to use for the API call
+//   - desiredWebhookConfig: The desired webhook configuration to create or update
+//
+// Returns:
+//   - error: Any error encountered while creating or updating the webhook configuration
+func createOrUpdateWebhookConfiguration(ctx context.Context, c client.Client,
```

Comment:
do we actually need a get and create|update ? is there any issue by using server side apply ?

Reply:
For the operator upgrade case, if we introduce a new webhook in the next version, the already existing webhook needs to be updated.

Reply:
I'm not sure if I see the relation with SSE, since SSE would create the webhook if it does not exist, would patch it otherwise.